src/v0/destinations/marketo_static_list/transformV2.js

```diff
@@ -0,0 +1,145 @@
+const lodash = require('lodash');
+const { InstrumentationError, UnauthorizedError } = require('@rudderstack/integrations-lib');
+const {
+  defaultPostRequestConfig,
+  defaultDeleteRequestConfig,
+  getDestinationExternalID,
+  defaultRequestConfig,
+  getSuccessRespEvents,
+  isDefinedAndNotNull,
+  generateErrorObject,
+  getErrorRespEvents,
+} = require('../../util');
+const { JSON_MIME_TYPE } = require('../../util/constant');
+const { MAX_LEAD_IDS_SIZE } = require('./config');
+const { getAuthToken } = require('../marketo/transform');
+const { formatConfig } = require('../marketo/config');
+
+/**
+ * Generates the final response structure to be sent to the destination
+ * @param {*} endPoint
+ * @param {*} leadIds
+ * @param {*} operation
+ * @param {*} token
+ * @returns batched response
+ */
+const responseBuilder = (endPoint, leadIds, operation, token) => {
+  let updatedEndpoint = endPoint;
+  if (leadIds.length > 0) {
+    leadIds.forEach((id) => {
+      updatedEndpoint = `${updatedEndpoint}id=${id}&`;
+    });
+  }
+  updatedEndpoint = updatedEndpoint.slice(0, -1);
+  const response = defaultRequestConfig();
+  response.endpoint = updatedEndpoint;
+  if (operation === 'insert') {
+    response.method = defaultPostRequestConfig.requestMethod;
+  } else {
+    response.method = defaultDeleteRequestConfig.requestMethod;
+  }
+  response.headers = {
+    Authorization: `Bearer ${token}`,
+    'Content-Type': JSON_MIME_TYPE,
+  };
+  return response;
+};
+
+/**
+ * The function is responsible for building the batched response for a given set of record inputs.
+ * @param {*} groupedRecordInputs
+ * @param {*} Config
+ * @param {*} token
+ * @param {*} leadIds
+ * @param {*} operation
+ * @returns an array of response objects, where each object represents a batched response for a chunk of lead IDs.
+ */
+const batchResponseBuilder = (listId, Config, token, leadIds, operation) => {
+  const { accountId } = Config;
+  const endpoint = `https://${accountId}.mktorest.com/rest/v1/lists/${listId}/leads.json?`;
+  const response = [];
+  const leadIdsChunks = lodash.chunk(leadIds, MAX_LEAD_IDS_SIZE);
+  leadIdsChunks.forEach((ids) => {
+    response.push(responseBuilder(endpoint, ids, operation, token));
+  });
+  return response;
+};
+
+/**
+ * A function that processes a list of grouped record inputs.
+ * It iterates through each input and groups the field IDs based on the action.
+ * @param {*} groupedRecordInputs
+ * @returns An array containing the batched responses for the insert and delete actions along with the metadata.
+ */
+async function processRecordInputs(groupedRecordInputs, destination) {
+  const token = await getAuthToken(formatConfig(destination));
```

Comment:
looks like this  call happens for every group, but it is only required once per destination id right? move this outside function

Reply:
This is a cached call, so once the token is stored in the cache there will be no subsequent calls made